#ver=1.4
---
monitorSetting:
    topicScanTime_ms : 1000.0 # Scan exist topics in every _ms
    subscribeMsgPack : "vehicle_interfaces"
    zedRGBTopicNameList : ["/zed/zed_node/left_raw/image_raw_color", "/zed/zed_node2/left_raw/image_raw_color"]
    zedDepthTopicNameList : ["/zed/zed_node/depth/depth_registered", "/zed/zed_node2/depth/depth_registered"]

RecordSetting:
    outputFilename : "datalog"
    samplingStep_ms : 10.0 # Sampling topic data in every _ms
    autoSaveTime_s : 60.0 # Save file in every _s
    recordTime_s : -1.0 # Total record time, non-stop if value < 0
    numOfImgSaveTh : 4 # Number of Image saving threads
    numOfGndSaveTh : 1 # Number of GroundDetect saving threads
    enabled_record : False

# Settings for Params class under vehicle_interfaces/params.h
generic_prop:
    # Node name and namespace. The namespace will automatically adding prefix at node name, topic name, service name, etc..
    namespace : "V0"
    nodeName : "dataserver"

    # Device ID, the ID will automatically formed the name of nodes and publisher topics. For instance, 
    # the node name will be formed as "<nodeName>_<id>_node", and the publisher topic name will be formed as "<topic_name>_<id>". 
    # Note: If the topic use to be subscriptions, then the topic name will not be formed.
    id : 0

    # Device information service configuration
    devInfoService : "" # Service name devinfo_0
    devInterface : "eth0" # eth0
    devMultiNode : True # False

    # QoS service configuration
    qosService : "qos_0" # Service name qos_0
    qosDirPath : "launch/qos"

    # Safety service configuration
    safetyService : "" # Service name safety_0

    # Time sync service configuration
    timesyncService : "timesync_0" # Service name timesync_0
    timesyncPeriod_ms : 600000.0 # 10 minute a sync for HPC. 2 minute a sync for sensors.
    timesyncAccuracy_ms : 2.0 # Restrict the duration of sync process time
    timesyncWaitService : True # Wait until time synced at construction.